### PR TITLE
No adapters uses `value.to_s` when `Type::Binary::Data` in `_quote`

### PR DESCRIPTION
**Pull request by kamipo**
All adapters (postgresql, mysql2, sqlite3, sqlserver, oracle-enhanced)
does not use `value.to_s` when `Type::Binary::Data` in `_quote`.

### CODE REVIEW

File: activerecord/lib/active_record/connection_adapters/sqlite3/quoting.rb [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+          quote_column_name(attr)
+        end
+
+        def quote_column_name(name)
+          @quoted_column_names[name] ||= %Q("#{name.to_s.gsub('"', '""')}")
+        end
+
+        private
+
+        def _type_cast(value)
```
Comment (egilburg) [inline activerecord/lib/active_record/connection_adapters/sqlite3/quoting.rb]: is this method related to quoting?

Comment (kamipo) [inline activerecord/lib/active_record/connection_adapters/sqlite3/quoting.rb]: This method inherits from [lib/active_record/connection_adapters/abstract/quoting.rb#L176-L189](https://github.com/rails/rails/blob/v5.0.0.beta2/activerecord/lib/active_record/connection_adapters/abstract/quoting.rb#L176-L189).

File: activerecord/lib/active_record/connection_adapters/abstract/quoting.rb [hunk 2]
```diff
@@ -158,8 +158,10 @@ def types_which_need_no_typecasting
 
       def _quote(value)
         case value
-        when String, ActiveSupport::Multibyte::Chars, Type::Binary::Data
+        when String, ActiveSupport::Multibyte::Chars
           "'#{quote_string(value.to_s)}'"
+        when Type::Binary::Data
+          "x'#{value.hex}'"
```
Comment (sgrif) [inline activerecord/lib/active_record/connection_adapters/abstract/quoting.rb]: This doesn't belong in the abstract adapter. It is not valid in the SQL standard.

**Comment by sgrif** [comment]
Thanks, but we shouldn't override the abstract adapter to do things that aren't valid for all adapters.
